x-pack/plugins/actions/server/routes/legacy/delete.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { IRouter } from 'kibana/server';
+import { ILicenseState, verifyApiAccess, isErrorThatHandlesItsOwnResponse } from '../../lib';
+import { BASE_ACTION_API_PATH } from '../../../common';
+import { ActionsRequestHandlerContext } from '../../types';
+
+const paramSchema = schema.object({
+  id: schema.string(),
+});
+
+export const deleteActionRoute = (
+  router: IRouter<ActionsRequestHandlerContext>,
+  licenseState: ILicenseState
+) => {
+  router.delete(
+    {
+      path: `${BASE_ACTION_API_PATH}/action/{id}`,
+      validate: {
+        params: paramSchema,
+      },
+    },
+    router.handleLegacyErrors(async function (context, req, res) {
+      verifyApiAccess(licenseState);
```

Comment:
I kept the legacy ones as is... we could reuse it here, but I didn't think it was worth rewriting all the tests for the unit tests legacy APIs. 🤔 

(I'e had feedback in the past to reduce change when it isn't necessary, and since we'll be SHIFT+DELETEing these files in a couple of minors, I wasn't too fussed).

I don't mind doing it though 🤷

Reply:
Gotcha that makes sense! It looks like the legacy `create` route does change it to `verifyAccessAndContext` and that is why I was wondering why the other legacy routes did not change it :)

Reply:
oh, oops, that's because that was the first one I migrated 🤦 
I now recall after having the pain of migrating it I realised this might not really be necessary 😆 

Thanks for flagging, but I think I'll leave it as is for now as we're deleting these _soon_ :D